Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.h

```diff
@@ -389,6 +393,19 @@ class TableInstance {
             if (max.value() < new_size)
                 return false;
         }
+        // FIXME: right now, since all i32's are turned into i64's, they're
+        // getting extended such that 2^32 becomes 2^64. This causes overflow
+        // on 64-bit platforms when adding to new_size (it's a size_t). All
+        // that to say that this check will probably never catch cases when
+        // the requested size is too large. The failing spec test is here:
+        // https://github.com/WebAssembly/testsuite/blob/6dfedc8b8423a91c1dc340d3af1a7f4fbf7868b4/table_grow.wast#L50
+        // table.grow works otherwise, but in order to fix this, we'll need to
+        // actually use 32-bit integers, not 64-bit ones, when representing
+        // i32's in WebAssembly. This will also fix a whole host of other
+        // sublte issues in the VM.
+        if (new_size >= 4294967296) {
```

Comment:
we...aren't storing i32's as i64? `Wasm::Value` has both i32 and i64.
This is really only because we're passing size_t to this particular function.

that aside, use `NumericLimits<u32>::max()` instead of the magic number pls.

Reply:
Oh shoot on further inspection you're totally right, my bad. I assumed it was the same as with floats, because I'm pretty sure all the f32's are getting stored as f64's, and I saw the cast [here](https://github.com/dzfrias/serenity/blob/3f879a3e63ac4d3e1e21f5318973884138bf1d2c/Userland/Libraries/LibWasm/AbstractMachine/BytecodeInterpreter.cpp#L515) and thought that was the case. I apologize for the misunderstanding, and I'll fix the magic number thing.